docs/Extensibility_UG/graph_rewrite_pass.md

```diff
@@ -3,7 +3,7 @@
 `ov::pass::GraphRewrite` serves for running multiple matcher passes on `ov::Model` in a single graph traversal.
 Example:
 
-@snippet src/transformations/template_pattern_transformation.cpp matcher_pass:graph_rewrite
+@snippet template_pattern_transformation.cpp matcher_pass:graph_rewrite
```

Comment:
why do we need this move?

Reply:
These transformation doesn't provide any benefits for template plugin. So I just moved them to snippets